### DISCUSSION TITLE
Help needed! [rust noob]

### DISCUSSION THREAD

**Discussion by YukieShiryuu** [jam1garner/binrw#128] [Q&A]
Good Night All,

I've been racking my brain trying to figure out how to parse a particular file all day. Here is an example,

`X...................X...........................X..............................X`

Where "X" is the delimiter and I'd like to capture all the data within the delimiters in a Vec<Vec<u8>>
Also from my observations this is somewhat like a binary csv format in a sense so the data is never even
i.e. one block can be 100 bytes the next could be 1000 bytes etc...

Issue 1 : I've tried quite a few of the attributes for #[br] to no avail
Issue 2 : I have an error about not having Vec<Vec<u8>> implemented so I think I need a trait ?
Issue 3 : I am very new to rust so I do apologize in advance for my ignorance and incompetence. 

Any help is welcome and thankyou.

**Comment by YukieShiryuu** [comment]
Ok so I think what I need is  `parse_with = until_eof_with`  that will populate a `Vec<Vec<u8>>` but currently I only have it working with `count(x)` how would I configure this to seek until the delimiter ? I see that it is possible with `until_with` but I cant figure it out.

Thanks

**Reply by YukieShiryuu** [reply]
Nice.

Got it kinda working with 

`    #[br(parse_with = until_eof_with(until(|&byte| byte == 88u8)))]
    value: Vec<Vec<u8>>,
`

but it is offset by 1 byte as i want to capture the 88 as well. I'll keep working on it ~

Thanks

**Reply by csnover** [reply]
Hey YukieShiryuu,

Can you give some more information about the data structure you are trying to process? Is the final form of the data really `Vec<Vec<u8>>` or is that just a starting point and you are planning on replacing the inner `Vec<u8>` type with something else later? Is the delimiter actually part of the preceding data, or do you just want to capture it separately for another purpose? Is the delimiter always a fixed value? There are a few different ways to parse correctly depending on these particulars; the `until` helpers (which you’ve already found) might be best if you have a straightforward termination condition; the [Punctuated](https://docs.rs/binrw/latest/binrw/punctuated/struct.Punctuated.html) type may be best if you are trying to parse interleaved data of two types; or maybe even just a plain function would be the easiest approach if there is some additional complexity that hasn’t been fully described here.

Thanks!
